fitbenchmarking/fitbenchmarking
# use residual, not function, in levmar

#### Description of Work

Fixes #804 by sending the residual, `eval_r` to levmar, rather than f(x,p) and y.  Since the residual is part of the cost function class, it is correct for the cost function selected.


#### Testing Instructions

1. Run against the three `nlls` cost functions and check default jacobian gives answers consistent with other methods (other jacobians may not, due to #776)
2. Run against the `poisson` cost function, and check that an error is returned

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
